File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cnt=0;
        int ele=0;
        for(int num:nums){
            if(cnt==0){
                ele=num;
                cnt++;
            }
            else if(num==ele)cnt++;
            else cnt--;
        }
        for(int num:nums){
            if(num==ele)return num;
        }
        return 0;
    }
}

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let ele=0;
    let cnt=0;
    for(const num of nums){
        if(cnt===0){
            ele=num;
            cnt++;
        }
        else if(ele===num)cnt++;
        else cnt--;
    }
    for(const num of nums){
        if(num===ele)return num;
    }
    return -1;
};